crates/paimon/src/spec/data_file.rs

```diff
@@ -90,16 +72,173 @@ pub struct DataFileMeta {
 }
 
 impl Display for DataFileMeta {
-    fn fmt(&self, _: &mut Formatter<'_>) -> std::fmt::Result {
-        todo!()
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{{fileName: {}, fileSize: {}, rowCount: {}, embeddedIndex: {:?}, \
+            minKey: {:?}, maxKey: {:?}, keyStats: {:?}, valueStats: {:?}, \
+            minSequenceNumber: {}, maxSequenceNumber: {}, \
+            schemaId: {}, level: {}, extraFiles: {:?}, creationTime: {}, deleteRowCount: {:?}}}",
+            self.file_name,
+            self.file_size,
+            self.row_count,
+            self.embedded_index,
+            self.min_key,
+            self.max_key,
+            self.key_stats,
+            self.value_stats,
+            self.min_sequence_number,
+            self.max_sequence_number,
+            self.schema_id,
+            self.level,
+            self.extra_files,
+            self.creation_time,
+            self.delete_row_count
+        )
     }
 }
 
 impl DataFileMeta {
-    // TODO: implement me
     pub const SCHEMA: RowType = RowType::new(vec![]);
+
+    /// Get the file name.
+    pub fn file_name(&self) -> &str {
+        &self.file_name
+    }
+
+    /// Get the file size.
+    pub fn file_size(&self) -> i64 {
+        self.file_size
+    }
+
+    /// Get the row count.
+    pub fn row_count(&self) -> i64 {
+        self.row_count
+    }
+
+    /// Get the min key.
+    pub fn min_key(&self) -> &BinaryRow {
+        &self.min_key
+    }
+
+    /// Get the max key.
+    pub fn max_key(&self) -> &BinaryRow {
+        &self.max_key
+    }
+
+    /// Get the key stats.
+    pub fn key_stats(&self) -> Option<&BinaryTableStats> {
+        self.key_stats.as_ref()
+    }
+
+    /// Get the value stats.
+    pub fn value_stats(&self) -> Option<&BinaryTableStats> {
+        self.value_stats.as_ref()
+    }
+
+    /// Get the min sequence number.
+    pub fn min_sequence_number(&self) -> i64 {
+        self.min_sequence_number
+    }
+
+    /// Get the max sequence number.
+    pub fn max_sequence_number(&self) -> i64 {
+        self.max_sequence_number
+    }
+
+    /// Get the schema id.
+    pub fn schema_id(&self) -> i64 {
+        self.schema_id
+    }
+
+    /// Get the level.
+    pub fn level(&self) -> i32 {
+        self.level
+    }
+
+    /// Get the extra files.
+    pub fn extra_files(&self) -> &[String] {
+        &self.extra_files
+    }
+
+    /// Get the creation time.
+    pub fn creation_time(&self) -> DateTime<Utc> {
+        self.creation_time
+    }
+
+    /// Get the delete row count.
+    pub fn delete_row_count(&self) -> Option<i64> {
+        self.delete_row_count
+    }
+
+    /// Get the embedded index.
+    pub fn embedded_index(&self) -> Option<&[u8]> {
+        self.embedded_index.as_deref()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_data_file_meta_serialize_deserialize() {
+        let json_data = r#"
```

Comment:
Hi, @devillove084

https://github.com/apache/paimon-rust/blob/main/crates/paimon/tests/fixtures/manifest_file_meta_schema.json

This fixture appears to be just the schema of `ManifestFileMeta`. How is `ManifestFileMeta` data stored on disk?

Reply:
Oh, we should also have `DataFileMeta`'s datatype and `DataFileMeta` on disk format.

Reply:
The default manifest file is of `avro` format. I think we should also generate a corresponding manifest file to verify the deserialize. But may be we should support avro/orc reader first ?